### PR TITLE
Allows the paypal_braintree source to be added to the wallet on checkout

### PR DESCRIPTION
**Pull request by joeljackson**

### CODE REVIEW

**Comment by joeljackson** [comment]
Looking for some feedback here. This is intended to address https://github.com/solidusio/solidus_paypal_braintree/issues/122.

Wanted to ensure a source is always reusable and that making it a subclass of Spree::PaymentSource wasn't going to ruin anyone's day before proceeding.

**Comment by swcraig** [comment]
Although it would make sense, I don't think that subclassing `Spree::PaymentSource` is the way to go here. `Spree::PaymentSource` [was only introduced in Solidus v2.2](https://github.com/solidusio/solidus/commit/c08f334e97cead1c10eb66fc1051e6d2a18b84f9), so subclassing it will [break this extension with all versions of Solidus before v2.2](https://travis-ci.org/solidusio/solidus_paypal_braintree/builds/270688825?utm_source=github_status&utm_medium=notification).

I believe your other two changes are what is needed to get this source added to the wallet.

**Comment by tvdeyen** [comment]
Unfortunately it’s needed to subclass from PaymentSource because Solidus introduced a validation on the payment source class in https://github.com/solidusio/solidus/blob/master/core/app/models/spree/wallet_payment_source.rb. 

We either should remove this validation (as this is useless IMO) or need to change this code so it only subclasses on Solidus 2.2 and above.

**Comment by joeljackson** [comment]
Seems like https://github.com/solidusio/solidus/pull/2201 should allow us to not subclass the PaymentSource. Will make appropriate changes.

**Comment by joeljackson** [comment]
Notably thinking about this it needs to be a subclass of PaymentSource on Solidus 2.3 or this wont

File: app/models/solidus_paypal_braintree/source.rb [hunk 1]
```diff
@@ -1,12 +1,13 @@
 module SolidusPaypalBraintree
-  class Source < ApplicationRecord
+  class Source < Spree::PaymentSource
```
Comment (jordan-brough) [inline app/models/solidus_paypal_braintree/source.rb]: Can you change this to `SolidusSupport.payment_source_parent_class` please?
See https://github.com/solidusio/solidus_support/pull/1

File: app/models/solidus_paypal_braintree/source.rb [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
     PAYPAL = "PayPalAccount"
     APPLE_PAY = "ApplePayCard"
     CREDIT_CARD = "CreditCard"
 
     belongs_to :user, class_name: Spree::UserClassHandle.new
     belongs_to :payment_method, class_name: 'Spree::PaymentMethod'
     has_many :payments, as: :source, class_name: "Spree::Payment"
+    has_many :wallet_payment_sources, class_name: 'Spree::WalletPaymentSource', as: :payment_source, inverse_of: :payment_source
```
Comment (jordan-brough) [inline app/models/solidus_paypal_braintree/source.rb]: You shouldn't need this if you inherit from `SolidusSupport.payment_source_parent_class`

**Comment by jordan-brough** [comment]
@joeljackson can you please also update the gemspec to require at least version `0.1.2` of solidus_support?

**Comment by joeljackson** [comment]
@jordan-brough Done.  Thanks 👍

**Comment by jordan-brough** [comment]
Thanks @joeljackson. Lgtm. Seeing if we can figure out what's wrong w/ the spec failures.

**Pull request merged**